lib/api/multiObjectDelete.js

```diff
@@ -166,6 +169,60 @@ function _parseXml(xmlToParse, next) {
     });
 }
 
+/**
+ * processObjectVersion - process object version to be deleted
+ * @param {object} entry - entry from data model
+ * @param {string} bucketName - bucket name
+ * @param {function} next - callback to call with error or decoded version
+ * @return {undefined}
+ **/
+function processObjectVersion(entry, bucketName) {
+    let decodedVersionId;
+    if (entry.versionId) {
+        decodedVersionId = entry.versionId === 'null' ?
+            'null' : versionIdUtils.decode(entry.versionId);
+    }
+    if (decodedVersionId instanceof Error) {
+        monitoring.promMetrics('DELETE', bucketName, 404,
+            'multiObjectDelete');
+        return [errors.NoSuchVersion];
+    }
+    return [null, decodedVersionId];
+}
+
+/**
+ * Initialization function for the MultiObjectDelete API that will, based on the
+ * current metadata backend,a ssess if metadata READ batching is supported. If
+ * yes, the initialization step will call the metadataGetObjects function from
+ * the MetadataWrapper.
+ * @param {string} bucketName - bucket name
+ * @param {string []} inPlay - list of object keys still in play
+ * @param {object} log - logger object
+ * @return {undefined}
+ */
+function shouldBatchGetObjectsFromMetadata(bucketName, inPlay, log) {
```

Comment:
It could be easier to read if the function returned a boolean, as one would expect from the function name, and just call the appropriate init function in the waterfall of `getObjectMetadataAndDelete`

Reply:
I would prefer keeping this function separately, that greatly ease the unit testing (and the base function is already quite complex :smile: ). Rather than just returning a flag, I will rename it `initializeMultiObjectDeleteWithBatchingSupport` instead, but I'm still open if you think we should change it anyway

Reply:
See https://github.com/scality/cloudserver/pull/5195/commits/9f7cc75b6259b61fcf0cbfa22d93fc544328f703